### ISSUE TITLE
Group association hability

### ISSUE THREAD

**Issue by alpaville21a14** [codahq/hubitat_codahq#7]
First of all thank you so much to develop this driver with the swipe function. I was wondering if you could add the hability of group association to this driver, so it will allow us to use the swipe for ou the 4 buttons. Unfortunately I have no skills to do it by myself but I found this on Inovelli's community and I think it can help a lot. https://community.inovelli.com/t/how-to-using-the-z-wave-association-tool-in-smartthings/1944/21. Thank you

**Comment by codahq** [comment]
Thanks for the link but I don't need it.  The driver already associates the swipe action to groups.  It associates the hub to all of the actions.  The thing I would need to do is change it to associate a node provided by the user via a preference to the group.  

The reason I didn't do it because it doesn't really work well to do anything.  If you associate, for example, a switch it won't do anything because most switches don't know what to do with a multi level start change command.  Most bulbs and dimmers handle these but I just couldn't imagine a scenario where a user would rather control the dim up and down of two bulbs individually rather than have a button event at the hub to do whatever they want.

I could make a preference for the remote to operate in direct association mode or in hub mode I guess.  

Do you really have 4 separate devices that support multi level start and stop change that you want to control separately in the same room as the remote?

**Comment by alpaville21a14** [comment]
Thanks for your answer. Actually I have 3, being 2 nano shutters and 1 nano
dimmer. Commonly I control them individually and, as we are in four people
here we have a lot of individual preferences and the shutters' positions
depends on the time of day, so the easiest way I thought would be if I
could use the group association. In this case I would be able to use swipe
and hold functions to control them separately and as desired.
If you are able to create this preference I would really appreciate.

Em qua, 27 de mai de 2020 21:41, codahq <notifications@github.com> escreveu:

**Comment by codahq** [comment]
Okay, I'll put it on the todo list.  No promises on time line.  Fala português?

**Comment by alpaville21a14** [comment]
Sim eu falo. Vc é brasileiro tb?

Em qua, 27 de mai de 2020 22:41, codahq <notifications@github.com> escreveu:

**Comment by codahq** [comment]
Não, sou americana, mas morava dois anos em Curitiba e aprendi a falar português.

**Comment by alpaville21a14** [comment]
Que legal! Curitiba é um ótimo lugar para viver, espero que você tenha
gostado de morar lá. Podemos conversar em português? Com certeza você fala
melhor português do que eu falo inglês hehe.
Eu ficaria muito feliz se você pudesse fazer essa modificação nesse device
handler, isso será muito útil para mim. Além de habilitar os swipes, será
muito útil também a habilidade de segurar o botão apertado e ele
automaticamente alternar entre dim up e dim down, coisa que hoje eu não
consigo fazer.

Você trabalha profissionalmente com esse tipo de programação ou é apenas um
hobbie?

Eu gosto muito da Aeotec, mas estou sofrendo um pouco para encontrar todos
os device handlers. O que não consigo encontrar de jeito nenhum é DH do
nano shutter. Você tem ideia de onde posso encontrá-lo ou tem planos para
fazê-lo? Eu comprei 4 mas ainda não sei como usá-los. Conversei com a
Aeotec e eles só tem o DH para hubitat que abre e fecha apenas, ou seja,
não tem a função de abrir ou fechar parcialmente (por exemplo abrir 30%).

Desde já muito obrigado pela ajuda

Rafael

Em dom, 31 de mai de 2020 04:34, codahq <notifications@github.com> escreveu:

**Comment by codahq** [comment]
Podemos conversar em português.  Eu gostei bastante de Curitiba e tenho saudades de vez em quando de churrascos e as pessoas que conhecia lá.  Faz muito tempo que retornei e não tenho contato mais com eles em geral.

Tudo isso é apenas um hobbie.  Nenhuma promessa na linha do tempo para essa modificação.  Enquanto isso, você pode definir os grupos com [isto](https://github.com/hubitat/HubitatPublic/blob/master/examples/drivers/basicZWaveTool.groovy) ou [aquilo](https://github.com/codahq/hubitat_codahq/blob/master/devicestypes/less-basic-z-wave-tool.groovy).

Eu dei uma olhada rápidinha e também não vejo um driver para a nano shutter.  Uma vez a Aoetec me deu um device de graça se eu escrever o driver para ele.  É possível que eles possam me dar um nano shutter.  Ainda está conversando com a Aeotec?  Diga a eles para me enviar um nano shutter.  Eles deveriam saber quem sou eu a menos que tenham um grupo diferent para o Brasil.

**Comment by codahq** [comment]
Talvez eles não devessem enviar um. Não tenho um motor de 24V.

**Comment by alpaville21a14** [comment]
Ele também funciona em motores 100-240V. Eu tenho um ticket aberto com eles
sobre este tema e eles estão dizendo que se eu usar o driver do nano dimmer
ele talvez funcione, então vou testar mas não acredito que irá funcionar
perfeitamente. Minha maior preocupação é que ele não funcione bem com as
aberturas parciais (segundo eles, se eu colocar o dimmer em X% o nano
shutter vai entender que é para abrir X%).
Eu vou falar com eles sobre você e pedir para que te mandem um nano
shutter, acredito que eles possam considerar.

Quanto ao wallmote quad, como eu devo utilizar o driver que vc enviou? Qual
parâmetro devo configurar e como? Desculpe as perguntas, é que sou meio
leigo no assunto e não quero correr o risco de fazer alguma besteira.

Obrigado

**Comment by alpaville21a14** [comment]
Boas noticias, falei com a Aeotec e eles vão te enviar o nano shutter. Eles
só pediram seu email para que eles entrem em contato direto com você.
Poderia por favor me passar?

Obrigado

Em seg, 1 de jun de 2020 09:11, Alphaville 21A14 <alphaville21a14@gmail.com>
escreveu:

> Ele também funciona em motores 100-240V. Eu tenho um ticket aberto com
> eles sobre este tema e eles estão dizendo que se eu usar o driver do nano
> dimmer ele talvez funcione, então vou testar mas não acredito que irá
> funcionar perfeitamente. Minha maior preocupação é que ele não funcione bem
> com as aberturas parciais (segundo eles, se eu colocar o dimmer em X% o
> nano shutter vai entender que é para abrir X%).
> Eu vou falar com eles sobre você e pedir para que te mandem um nano
> shutter, acredito que eles possam considerar.
>
> Quanto ao wallmote quad, como eu devo utilizar o driver que vc enviou?
> Qual parâmetro devo configurar e como? Desculpe as perguntas, é que sou
> meio leigo no assunto e não quero correr o risco de fazer alguma besteira.
>
> Obrigado
>
